### PR TITLE
Create card space on merchant create

### PR DESCRIPTION
**Pull request by jurgenwerk**
Ticket: [CS-3138](https://linear.app/cardstack/issue/CS-3138/when-business-account-is-created-create-a-card-space-associated-with)

With this, every created merchant will automatically get a card space.

### CODE REVIEW

File: packages/hub/routes/merchant-infos.ts [hunk 1]
```diff
@@ -90,10 +95,22 @@ export default class MerchantInfosRoute {
       ownerAddress: ctx.state.userAddress,
     };
 
-    await this.merchantInfoQueries.insert(merchantInfo);
+    let db = await this.databaseManager.getClient();
+    let merchantInfoId, cardSpaceId;
+
+    await inTransaction(db, async () => {
```
Comment (jurgenwerk) [inline packages/hub/routes/merchant-infos.ts:101]: This is so that we don't end up with a merchant without a card space, in case card space insert fails.

**Review by backspace — APPROVED** [review]

**Review by Aierie — APPROVED** [review]

File: packages/hub/routes/merchant-infos.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/hub/routes/merchant-infos.ts:101]: Good idea! Will put it there

File: packages/hub/utils/queries.ts [hunk 2]
```diff
@@ -36,3 +37,20 @@ export function buildConditions(params: any, tableName?: string) {
     values: Object.values(nonNullParams),
   };
 }
+
+export async function inTransaction(db: Client, cb: any) {
+  let isTestEnv = process.env.NODE_ENV === 'test';
+
+  if (isTestEnv) {
+    await cb(); // Don't commit any data to the test db
```
[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/hub/utils/queries.ts:45]: The queries that run in tests are running in a transaction that is rolled back when tests finish. I tested this and the nested transaction still gets committed to the DB, even when the outer transaction gets rolled back! It turns out the correct way to support nested transaction is by using the [SAVEPOINT syntax](https://stackoverflow.com/questions/25425944/does-postgres-support-nested-or-autonomous-transactions).

I thought supporting nested transactions is a bit of an overkill for now so I decided to turn it off in tests where we know the queries are already running in a transaction. Might be good to add a comment that this method only works when there is no outer transaction

[the remark this answers is not in the document]
Comment (jurgenwerk) [reply packages/hub/utils/queries.ts:45]: Or, maybe I can use the savepoint syntax as a drop in replacement for the `BEGIN`/`COMMIT` and both nested and unnested transactions will work as intended, without disabling it for tests! Will try it out, thanks!

**Pull request merged**